internal/olm/operator/registry/configmap/deployment_test.go

```diff
@@ -0,0 +1,256 @@
+			}
+			f := withContainerVolumeMounts("testVolName", paths...)
+			f(dep)
+
+			Expect(dep.Spec.Template.Spec.Containers).Should(Equal(expecteddep.Spec.Template.Spec.Containers))
+		})
+	})
+
+	Describe("getDBContainerCmd", func() {
+		It("should apply f to dep's pod template spec and apply volumes", func() {
+			initCmd := "/bin/initializer -o /path/to/database.db -m /registry/manifests"
+			srvCmd := "/bin/registry-server -d /path/to/database.db -t /var/log/temp.log"
+
+			Expect(getDBContainerCmd("/path/to/database.db", "/var/log/temp.log")).Should(Equal(fmt.Sprintf("%s && %s", initCmd, srvCmd)))
+		})
+	})
+
+	Describe("withRegistryGRPCContainer", func() {
+		It("should apply f to dep's pod template spec and append contaiers", func() {
+			container := corev1.Container{
+				Name:       getRegistryServerName("testPkg"),
+				Image:      registryBaseImage,
+				WorkingDir: "/tmp",
+				Command:    []string{"/bin/sh"},
+				Args: []string{
+					"-c",
+					getDBContainerCmd(registryDBName, registryLogFile),
+				},
+				Ports: []corev1.ContainerPort{
+					{Name: "registry-grpc", ContainerPort: registryGRPCPort},
+				},
+			}
+			dep := &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Replicas: nil,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							NodeName:   "testNode",
+							Containers: nil,
+						},
+					},
+				},
+			}
+			expecteddep := &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Replicas: nil,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							NodeName:   "testNode2",
+							Containers: nil,
+						},
+					},
+				},
+			}
+
+			expecteddep.Spec.Template.Spec.Containers = append(expecteddep.Spec.Template.Spec.Containers, container)
+			f := withRegistryGRPCContainer("testPkg")
+			f(dep)
+
+			Expect(dep.Spec.Template.Spec.Containers).Should(Equal(expecteddep.Spec.Template.Spec.Containers))
+		})
+	})
+
+	Describe("newRegistryDeployment", func() {
+		var (
+			replicas int32
+			dep      *appsv1.Deployment
+		)
+		BeforeEach(func() {
+			replicas = 1
+			dep = &appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: appsv1.SchemeGroupVersion.String(),
+					Kind:       "Deployment",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      getRegistryServerName("testPkg"),
+					Namespace: "testns",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: getRegistryDeploymentLabels("testPkg"),
+					},
+					Replicas: &replicas,
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: getRegistryDeploymentLabels("testPkg"),
+						},
+					},
+				},
+			}
+		})
+		It("should return a dployment", func() {
+			f := withRegistryGRPCContainer("testPkg")
+			f(dep)
+
+			Expect(dep).Should(Equal(newRegistryDeployment("testPkg", "testns", f)))
+		})
+		It("should return a dployment for a custom made function", func() {
```

Comment:
I don't quite understand the expected behavior if len(opts) > 1, but that should also be tested.